scripted/sbt/src/main/scala/sbt/test/ScriptedTests.scala

```diff
@@ -108,19 +109,27 @@ object ScriptedTests {
     val logger = ConsoleLogger()
     run(directory, buffer, tests, logger, bootProperties, Array(), emptyCallback)
   }
+}
+
+class ScriptedRunner {
+  import ScriptedTests._
+
   def run(resourceBaseDirectory: File, bufferLog: Boolean, tests: Array[String], bootProperties: File,
     launchOpts: Array[String]): Unit =
     run(resourceBaseDirectory, bufferLog, tests, ConsoleLogger(), bootProperties, launchOpts, emptyCallback) //new FullLogger(Logger.xlog2Log(log)))
 
+  // This is called by project/Scripted.scala
+  // Using java.util.List[File] to encode File => Unit
   def run(resourceBaseDirectory: File, bufferLog: Boolean, tests: Array[String], bootProperties: File,
-    launchOpts: Array[String], prescripted: File => Unit): Unit =
-    run(resourceBaseDirectory, bufferLog, tests, ConsoleLogger(), bootProperties, launchOpts, prescripted) //new FullLogger(Logger.xlog2Log(log)))
+    launchOpts: Array[String], prescripted: java.util.List[File]): Unit =
+    run(resourceBaseDirectory, bufferLog, tests, ConsoleLogger(), bootProperties, launchOpts,
+      { f: File => prescripted.add(f); () }) //new FullLogger(Logger.xlog2Log(log)))
 
-  def run(resourceBaseDirectory: File, bufferLog: Boolean, tests: Array[String], logger: AbstractLogger, bootProperties: File,
+  def run(resourceBaseDirectory: File, bufferLog: Boolean, tests: Array[String], logger: xsbti.Logger, bootProperties: File,
```

Comment:
I can add shims around to keep bincompat, but this module should be used only by sbt itself, so I don't think it matters.